packages/app/src/lib/import.ts

```diff
@@ -48,49 +55,49 @@ export function loadPapa(): Promise<any> {
 /**
  * Takes a data table (represented by a two-dimensional array) and converts it
  * into an array of items
- * @param  Array    data         Two-dimensional array containing tabular item data; The first 'row'
- *                               should contain field names. All other rows represent items, containing
- *                               the item name, field values and optionally a list of tags.
- * @param  Integer  nameColIndex Index of the column containing the item names. Defaults to 0
- * @param  Integer  tagsColIndex  Index of the column containing the item categories. If left empty
- *                               no categories will be used
+ * @param  Array    data            Two-dimensional array containing tabular item data; The first 'row'
+ *                                  should contain field names. All other rows represent items, containing
+ *                                  the item name, field values and optionally a list of tags.
+ * @param  Array    columnTypes     Array containing the type of field per column.
+ * @param  Boolean  dataOnFirstRow  Boolean, representing if there is data on the first row (instead of
+ *                                  columns).
  */
-export async function fromTable(data: string[][], nameColIndex?: number, tagsColIndex?: number): Promise<VaultItem[]> {
-    // Use first row for column names
-    const colNames = data[0];
-
-    if (nameColIndex === undefined) {
-        const i = colNames.indexOf("name");
-        nameColIndex = i !== -1 ? i : 0;
+async function fromTable(
+    data: string[][],
+    columnTypes: ImportCSVColumn[],
+    dataOnFirstRow: boolean
```

Comment:
Minor nitpick, but I think the reverse makes more sense to me somehow (probably because that's what I'm used from other apps). I.e. `firstRowHasFieldNames`.

Reply:
Naming things! I personally don't like _true defaults_, but I can go with that and default to having the toggle on.

Reply:
> Naming things!

Hahah, yeah... Well, the naming of the variable in the code I'll leave up to you, but for the user-facing option I do think "First row contains field names" is the clearer naming. Why? Because if you write "First row contains data" the natural question is: "What else would it contain?". Whereas "contains field names" makes it clear that if the option is disabled, the row will be treated as all other rows. Does that make sense?